Add build options to the srtool build step

EgorPopelyaev opened the pull request:
This PR adds possibility to set BUILD_OPTIONS to the "Srtool Build"  step in the release pipeline while building runtimes.

Colses: https://github.com/paritytech/release-engineering/issues/213

ggwpez commented on .github/workflows/release-srtool.yml:
```diff
@@ -76,9 +78,20 @@ jobs:
+        run: |
+          BUILD_OPTIONS=""
+          if [[ ${{ matrix.chain }} == 'asset-hub-rococo' || ${{ matrix.chain }} == 'asset-hub-westend' || ${{ matrix.chain }} == 'rococo' || ${{ matrix.chain }} == 'westend' ]]; then
+            BUILD_OPTIONS="${{ inputs.build_opts }}"
```
Conceptually it seems a bit odd that it ignores the build opts for some runtimes. Maybe you can add a `METADATA_HASH_ENABLED` flag or something?

bkchr replied:
We should add `on-chain-release-build` to every runtime and use this one here.

EgorPopelyaev replied:
This was the initial plan, the reason why I went this way was, that it includes this feature `sp-api/disable-logging` that turns off the logging, that is on the other hand needed for devops guys cc: @PierreBesson. After discussion with Pierre, that was kinda compromised solution🙈 to not build two types of runtimes (with on chain feature and with logging). If there is another to do it, I'll be happy to use it:). Otherwise, I can also add additional step to build those runtimes with logs for devops.

bkchr replied:
We are also running logging nodes for main net, because we have logging disabled there. However, we can also just remove the "disable-logging" feature there.

EgorPopelyaev replied:
I've tested the solution with two separate jobs, building runtimes with and without logging (having `on-chain-release-build` option). It looks fine. The only thing now is that `people-rococo` and `poepele-westend` seems not have this feature. Whom could I contact to ask to add this option? Is it Joe?

bkchr replied:
You can just copy the feature declaration from the other runtimes.

bkchr replied:
And then we should remove the `disable-logging` everywhere.

EgorPopelyaev replied:
Basti, I've added the `on-chain-release-build` where it was missing, should the `disable-logging` deletion be done here as well or as a separate task? I'm not sure if it is worth to mention it in the rel notes or not.

ggwpez commented:
I think we should generally re-evaluate if we still need srtool at all, since we should not in theory.  
But since the current release process relies on it, its probably fine to do this later.

EgorPopelyaev commented:
> I think we should generally re-evaluate if we still need srtool at all, since we should not in theory. But since the current release process relies on it, its probably fine to do this later.

Yep, I agree with that, we can do it step by step. After the switch to the new process, we can start to re-think what is needed for the current setup and what can be replaced

bkchr commented:
> I think we should generally re-evaluate if we still need srtool at all, since we should not in theory. But since the current release process relies on it, its probably fine to do this later.

`srtool` only exist to have reproducible builds. There is no other reason for its existence. I'm not sure what the current state of rust and reproducible builds is. If we don't need it anymore, we can remove it.

bkchr commented on .github/workflows/release-30_publish_release_draft.yml:
```diff
@@ -26,6 +29,15 @@ jobs:
+    uses: "./.github/workflows/release-srtool.yml"
+    with:
+       excluded_runtimes: "substrate-test bp cumulus-test kitchensink minimal-template parachain-template penpal polkadot-test seedling shell frame-try sp solochain-template"
+       build_opts: ""
```
This doesn't enable the `on-chain-release-build` feature. This means that metadata hash will not be generated at compile time. This is bad :see_no_evil:

EgorPopelyaev replied:
Yes, this one was done only for the case to have the logging enabled. In theory, this step should go away if the `disable-logging` feature will be deleted from the `on-chain-release-build` as we discussed above. If it is ok, I can delete it as part of this PR, and then we will have only the one build with the `on-chain-release-build` feature activated.

EgorPopelyaev commented:
@bkchr @ggwpez Basti, Oli, I've deleted the `disable-logging` from the `on-chain-release-build` feature as discussed [above](https://github.com/paritytech/polkadot-sdk/pull/4956#discussion_r1669278563) and adjusted the flow so, that we build the runtimes only once with `on-chain-release-build` activated. Could you please have a look if it is fine so and PR can be merged?
